### ISSUE TITLE
Argument list too long issue

### ISSUE THREAD

**Issue by rpalcolea** [spotbugs/spotbugs-gradle-plugin#239]
Hi folks,

Found that having projects with massive classpaths can not be used with spotbugs plugin. For example:

```
17:29:09 Caused by: org.gradle.process.internal.ExecException: Process 'command '/usr/lib/jvm/zulu-8-amd64/bin/java'' could not be started because the command line exceed operating system limits.
17:29:09 	at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:241)
17:29:09 	at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:218)
17:29:09 	at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:372)
17:29:09 	at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:87)
17:29:09 	at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
17:29:09 	... 3 more
17:29:09 Caused by: net.rubygrapefruit.platform.NativeException: Could not start '/usr/lib/jvm/zulu-8-amd64/bin/java'
17:29:09 	at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27)
17:29:09 	at net.rubygrapefruit.platform.internal.WrapperProcessLauncher.start(WrapperProcessLauncher.java:36)
17:29:09 	at org.gradle.process.internal.ExecHandleRunner.startProcess(ExecHandleRunner.java:98)
17:29:09 	at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:71)
17:29:09 	... 4 more
17:29:09 Caused by: java.io.IOException: Cannot run program "/usr/lib/jvm/zulu-8-amd64/bin/java" (in directory "/agent/workspace/myproject"): error=7, Argument list too long
17:29:09 	at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
17:29:09 	... 7 more
17:29:09 Caused by: java.io.IOException: error=7, Argument list too long
```

While path jar is supported in gradle and can be overwritten via `      System.setProperty("org.gradle.internal.cmdline.max.length", CMDLINE_MAX_LENGTH);`, it is not enough since the classpath for the jar is small as it contains only the spotbugs/findbugs jars

 https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/process/internal/util/LongCommandLineDetectionUtil.java#L45

which is invoked by https://github.com/gradle/gradle/blob/9e347dcbd00483d796e83c05c8b48d61dad06d0c/subprojects/core/src/main/java/org/gradle/process/internal/JavaExecHandleBuilder.java#L388 to determine if it should do path jar or not

However, the problem comes with java processes trying to use a long `auxclasspath`.

This is a sample of simple project:

```
0 = "-sortByClass"
1 = "-timestampNow"
2 = "-auxclasspath"
3 = "/Users/rperezalcolea/.gradle/caches/modules-2/files-2.1/com.google.guava/guava/19.0/6ce200f6b23222af3d8abb6b6459e6c44f4bb0e9/guava-19.0.jar"
4 = "-sourcepath"
5 = "/Users/rperezalcolea/Projects/bb/nebula/nebula/subprojects/core/build/nebulatest/netflix.nebula.spotbugs.NebulaSpotbugsCiPluginLauncherSpec/should-produce-html-reports-by-default/src/main/resources:/Users/rperezalcolea/Projects/bb/nebula/nebula/subprojects/core/build/nebulatest/netflix.nebula.spotbugs.NebulaSpotbugsCiPluginLauncherSpec/should-produce-html-reports-by-default/src/main/java"
6 = "-html"
7 = "-outputFile"
8 = "/Users/rperezalcolea/Projects/bb/nebula/nebula/subprojects/core/build/nebulatest/netflix.nebula.spotbugs.NebulaSpotbugsCiPluginLauncherSpec/should-produce-html-reports-by-default/build/reports/spotbugs/main.html"
9 = "-projectName"
10 = "should-produce-html-reports-by-default (spotbugsMain)"
11 = "-release"
12 = "unspecified"
13 = "-analyzeFromFile"
14 = "/var/folders/hb/c0ghc68d7vn26h4ny84kzv0r0000gn/T/spotbugs-gradle-plugin3618879618624022328.txt"
```

When these arguments are longer than `ARG_MAX` (2097152 on linux / 262144 OSX / 32767 windows) the process will fail.

Wondering if there is a way in spotbugs to provide a file/manifest with the real classpath and work similar to what a path jar would do. It seems like it would also be nice default behavior to prevent this issue

Thanks

**Comment by rpalcolea** [comment]
actually, looking at https://github.com/spotbugs/spotbugs/blob/5f0246625b8f84500a76b3cff4991a9d6fe9a60c/spotbugs/src/main/java/edu/umd/cs/findbugs/TextUICommandLine.java#L212 it seems that this is possible.

Wonder if the plugin should convert the auxclasspath into a temporary file and pass the file instead either as default or under certain circumstances such as long argument

**Comment by KengoTODA** [comment]
closed by #243
